src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/SAM/SAMUtils.cs

```diff
@@ -94,8 +94,6 @@ internal static bool GetOSVersion(DirectoryEntry computerDE, out int versionMajo
 
         internal static Principal DirectoryEntryAsPrincipal(DirectoryEntry de, StoreCtx storeCtx)
         {
-            string className = de.SchemaClassName;
```

Comment:
SchemaClassName has side-effects. Why is it safe to remove?

Reply:
I am not exactly sure if the side effect is required or not in this particular case so I am just going to revert this change.